src/v/cluster/rm_stm.cc

```diff
@@ -2605,4 +2624,79 @@ ss::future<> rm_stm::handle_eviction() {
       });
 }
 
+void rm_stm::spawn_background_clean_for_pids(rm_stm::clear_type type) {
```

Comment:
Should we also lock mutex here to spawn only one clear process per time?

Reply:
yep, that's a good idea